graph/run-dynamic.go

```diff
@@ -187,7 +187,7 @@ func ConvergeDynamicThread[EP EPP[E], V VPI[V], E EPI[E], M MVI[M], N any, A Alg
 		}
 
 		// The main check for updates to topology. This occurs with priority over algorithmic messages.
-		if !strucClosed && !blockTop {
+		if !epoch && !strucClosed && !blockTop {
```

Comment:
I think we can also skip edge events when waiting for a query to finish? This if branch is very slow when there's no edge events due to backoff.

Reply:
Hm, I don't think this is okay -- a thread does need to ensure it has first drained anything in the topology queue.
But we can add another bool flag -- if a thread has epoch=true, and then has confirmed count = 0 from here, it can set that flag to true.
Then it can skip this loop from then on (and reset that flag when resetting epoch)

Reply:
Oh, why does a thread need to ensure it has drained the topology queue first?

Reply:
Currently the remitter is semantically the "user", and provides the query request. The remitter also is who put events into the TopologyQueue.
When a thread first observes the request for a query it is not guaranteed to have drained its TopologyQueue -- it should do that to ensure the view of the graph topology when providing a result back to the "user" is the same as the user expects.

There could be an option for async queries, but by default we should assume the query should be synced with the topology provided.

Reply:
That makes sense. Thank you!